daemon/src/taker_cfd.rs

```diff
@@ -766,22 +720,39 @@ where
             wire::MakerToTaker::RejectRollOver(order_id) => {
                 log_error!(self.handle_roll_over_rejected(order_id))
             }
-            MakerToTaker::RollOverProtocol(roll_over_msg) => {
+            wire::MakerToTaker::RollOverProtocol(roll_over_msg) => {
                 log_error!(self.handle_inc_roll_over_msg(roll_over_msg))
             }
+            wire::MakerToTaker::Heartbeat => {
+                unreachable!("Heartbeats should be handled somewhere else")
+            }
+            wire::MakerToTaker::ConfirmOrder(_)
+            | wire::MakerToTaker::RejectOrder(_)
+            | wire::MakerToTaker::Protocol { .. } => {
+                unreachable!("These messages should be sent to the `setup_taker::Actor`")
+            }
         }
     }
 }
 
 #[async_trait]
-impl<O: 'static, M: 'static, W: 'static> Handler<CfdSetupCompleted> for Actor<O, M, W>
+impl<O: 'static, M: 'static, W: 'static> Handler<setup_taker::Completed> for Actor<O, M, W>
 where
     O: xtra::Handler<oracle::MonitorAttestation>,
     M: xtra::Handler<monitor::StartMonitoring>,
     W: xtra::Handler<wallet::TryBroadcastTransaction>,
 {
-    async fn handle(&mut self, msg: CfdSetupCompleted, _ctx: &mut Context<Self>) {
-        log_error!(self.handle_cfd_setup_completed(msg.order_id, msg.dlc));
+    async fn handle(&mut self, msg: setup_taker::Completed, _ctx: &mut Context<Self>) {
+        use setup_taker::Completed::*;
+        match msg {
+            NewContract { order_id, dlc } => {
+                log_error!(self.handle_new_contract(order_id, dlc))
+            }
+            Rejected { order_id } => log_error!(self.handle_order_rejected(order_id)),
+            Failed { order_id, error } => {
+                log_error!(self.handle_contract_setup_failed(order_id, error))
+            }
+        }
```

Comment:
Actually, there is one more thing.

Can we _not_ split `Completed` into individual handlers here? With our new model, we will pass the entire result to the aggregate and no longer make this decision in the actor.

Reply:
Sorry for only noticing this now!

Reply:
@thomaseizinger @da-kami: I think this is what you guys meant, but the difference is pretty small, so maybe you were referring to something else: https://github.com/itchysats/itchysats/pull/684/commits/d7f7d8f91292e11edea561832aafb3e170816d59

Reply:
Nah that is perfect, thank you!